### PR TITLE
feat: add metrics to engine RPC API

### PR DESCRIPTION
**Pull request by Rjected**
This adds metrics to all engine API endpoints. Looking for feedback on whether or not these should be split up into separate `success` or `error` metrics, and whether or not the following APIs should be included:
* `engine_exchangeCapabilitiesV1`
* `engine_exchangeTransitionConfigurationV1`

### CODE REVIEW

File: crates/rpc/rpc-engine-api/src/engine_api.rs [hunk 1]
```diff
@@ -736,13 +775,21 @@ where
...
+        self.metrics.exchange_transition_configuration.record(start.elapsed());
+        Ok(res?)
     }
 
     /// Handler for `engine_exchangeCapabilitiesV1`
     /// See also <https://github.com/ethereum/execution-apis/blob/6452a6b194d7db269bf1dbd087a267251d3cc7f8/src/engine/common.md#capabilities>
     async fn exchange_capabilities(&self, _capabilities: Vec<String>) -> RpcResult<Vec<String>> {
-        Ok(CAPABILITIES.into_iter().map(str::to_owned).collect())
+        let start = Instant::now();
+        // TODO: might not be worth including this
```
Comment (mattsse) [inline crates/rpc/rpc-engine-api/src/engine_api.rs:789]: including this in what?

this is a constant, so probably not particularly useful

File: crates/rpc/rpc-engine-api/src/engine_api.rs [hunk 2]
```diff
@@ -33,6 +34,7 @@ const MAX_PAYLOAD_BODIES_LIMIT: u64 = 1024;
 /// functions in the Execution layer that are crucial for the consensus process.
 pub struct EngineApi<Provider> {
     inner: Arc<EngineApiInner<Provider>>,
+    metrics: EngineApiMetrics,
```
Comment (mattsse) [inline crates/rpc/rpc-engine-api/src/engine_api.rs:37]: this should be moved to the inner type

**Review by mattsse — APPROVED** [review]
one smol nit and q re todo

File: crates/rpc/rpc-engine-api/src/engine_api.rs (hunk 1 above)
Comment (Rjected) [reply crates/rpc/rpc-engine-api/src/engine_api.rs:789]: included as in, should we even have a metric for it at all, will remove

Comment (mattsse) [reply crates/rpc/rpc-engine-api/src/engine_api.rs:789]: gotcha,

I don't think it's very useful

File: crates/rpc/rpc-engine-api/src/engine_api.rs (hunk 2 above)
Comment (Rjected) [reply crates/rpc/rpc-engine-api/src/engine_api.rs]: moved

Comment (Rjected) [reply crates/rpc/rpc-engine-api/src/engine_api.rs]: moved

**Review by mattsse — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]
we can do err/success rates separately, i think latency works as is

**Pull request merged**